### PR TITLE
feat(remote-development): added remote-sshfs plugin

### PR DESCRIPTION
**Pull request by WilliamBy**
## 📑 Description

- Add plugin [remote-sshfs.nvim](https://github.com/nosduco/remote-sshfs.nvim)
- Require `sshfs` & `ssh`
- Bind keymap for convenience:
    - `<leader>Re` edit ssh config file
    - `<leader>Rc` select ssh target which is saved in your ssh config file
    - `<leader>Rd` disconnect (this will unmount sshfs)
- Override telescope find_files and live_grep to make dynamic based on if connected to host
    - Need `fd` and `ripgrep` be installed on server machine
- Check plugin README for more information

## ℹ Additional Information

- Require `sshfs` & `ssh` on `$PATH`

### CODE REVIEW

**Comment by Uzaaft** [comment]
Please fix the CI

File: lua/astrocommunity/remote-development/remote-sshfs/init.lua [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+				function()
+					if connections.is_connected then
+						api.live_grep()
+					else
+						builtin.live_grep()
+					end
+				end,
+				desc = "Find files",
+			}
+		end,
```
Comment (mehalter) [inline lua/astrocommunity/remote-development/remote-sshfs/init.lua:45]:
```suggestion
    opts = function(_, opts)
      local map = opts.mappings
      map.n["<Leader>R"] = { desc = "Remote" }
      map.n["<Leader>Rc"] = { function() require("remote-sshfs.api").connect() end, desc = "Connect" }
      map.n["<Leader>Rd"] = { function() require("remote-sshfs.api").disconnect() end, desc = "Disconnect" }
      map.n["<Leader>Re"] = { function() require("remote-sshfs.api").edit() end, desc = "Edit" }
      -- Override telescope find_files and live_grep to make dynamic based on if connected to host
      local find_files = map.n["<Leader>ff"]
      if type(find_files) == "table" then
        local orig_find_files = find_files[1]
        find_files[1] = function()
          if require("remote-sshfs.connections").is_connected then
            require("remote-sshfs.api").find_files()
          elseif type(orig_find_files) == "function" then
            orig_find_files()
          elseif type(orig_find_files) == "string" then
            vim.cmd(orig_find_files)
          end
        end
      end
      local live_grep = map.n["<Leader>fw"]
      if type(live_grep) == "table" then
        local orig_live_grep = live_grep[1]
        live_grep[1] = function()
          if require("remote-sshfs.connections").is_connected then
            require("remote-sshfs.api").live_grep()
          elseif type(orig_live_grep) == "function" then
            orig_live_grep()
          elseif type(orig_live_grep) == "string" then
            vim.cmd(orig_live_grep)
          end
        end
      end
    end,
```

Fixes bugs in lazy loading and improve robustness of overriding original mappings

**Review by mehalter — CHANGES_REQUESTED** [review]
Apply `stylua` after applying the requested changes to make sure the formatting is correct.

File: lua/astrocommunity/remote-development/remote-sshfs/init.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---@type LazySpec
+return {
+	"nosduco/remote-sshfs.nvim",
```
Comment (mehalter) [inline lua/astrocommunity/remote-development/remote-sshfs/init.lua:3]:
```suggestion
  "nosduco/remote-sshfs.nvim",
  cmd = {
    "RemoteSSHFSConnect",
    "RemoteSSHFSDisconnect",
    "RemoteSSHFSEdit",
    "RemoteSSHFSFindFiles",
    "RemoteSSHFSLiveGrep",
  },
```

add lazy loading on command

**Review by mehalter — APPROVED** [review]
Resolved issues with spec, looks great! Thanks for the contribution!

**Pull request merged**